### DISCUSSION TITLE
Handling mis-identification.

### DISCUSSION THREAD

**Discussion by ThisIsTheOnlyUsernameAvailable** [blakeblackshear/frigate#2995] [Ask A Question] [answered]
Hi!

Firstly, what an amazing project.  I got my first camera setup today and am blown away by the quality of the software and its functionality.  Thank you so much for your skills and effort.

I'm a little overwhelmed with the software so am trying to get my head around how to manage it most effectively.  I have had an event trigger on what I think is a combination of late afternoon sunlight / moving shadows and a pair of umbrellas that probably appear like legs:

![snapshot-1647842828 00832-fsuevg](https://user-images.githubusercontent.com/18511993/159215669-08b659e9-3ab4-46c3-bf61-608c18b349a8.jpg)

The detection is rated at 55%, and the event's score at 76.95%.

I can't mask the area as that umbrella stand is right next to the door, and Mrs would not allow redecorating.

So I guess my questions are:

1) Can I train the recognition algorithm to not detect the umbrellas, regardless of moving lighting?
2) Is it just a case of increasing the detection threshold to, say, 60%, to reduce false positives?
3) Why did the event score reach ~77% when the confidence of detection was 55%?

Many thanks.

**Comment by blakeblackshear** [comment, marked as answer]
> Can I train the recognition algorithm to not detect the umbrellas, regardless of moving lighting

Not easily. This something I am hoping to address with Frigate+.

> Is it just a case of increasing the detection threshold to, say, 60%, to reduce false positives

Filters would be your first option. I don't think you can filter that out with min_area. You could try raising min_score, but you may miss some true positives. I think this is a case where you need a small object filter mask around the bottom center of that bounding box for the stand.

> Why did the event score reach ~77% when the confidence of detection was 55%

The snapshot is a single frame of an event. The score is the top score for the object across all frames where it has been tracked.

**Comment by ThisIsTheOnlyUsernameAvailable** [comment]
Thanks so much for the advice!
